utils/resultstable.go

```diff
@@ -265,15 +266,35 @@ func prepareVulnerabilities(vulnerabilities []services.Vulnerability, results *R
 	return vulnerabilitiesRows, nil
 }
 
+// sortVulnerabilityOrViolationRows is sorting in the following order:
+// Severity -> Applicability -> JFrog Research Score -> XRAY ID
 func sortVulnerabilityOrViolationRows(rows []formats.VulnerabilityOrViolationRow) {
 	sort.Slice(rows, func(i, j int) bool {
 		if rows[i].SeverityNumValue != rows[j].SeverityNumValue {
 			return rows[i].SeverityNumValue > rows[j].SeverityNumValue
 		}
-		return len(rows[i].FixedVersions) > 0 && len(rows[j].FixedVersions) > 0
+		if rows[i].Applicable != rows[j].Applicable {
+			return jasutils.ConvertApplicableToScore(rows[i].Applicable) > jasutils.ConvertApplicableToScore(rows[j].Applicable)
+		}
+		priorityI := getJfrogResearchPriority(rows[i].JfrogResearchInformation)
+		priorityJ := getJfrogResearchPriority(rows[j].JfrogResearchInformation)
+		if priorityI != priorityJ {
+			return priorityI > priorityJ
+		}
+		return rows[i].IssueId > rows[j].IssueId
 	})
 }
 
+// getJfrogResearchPriority returns the score of JFrog Research Severity.
+// If there is no such severity will return the const value noJfrogResearchInformation which is less than all possible scores
+func getJfrogResearchPriority(jfrogResearchInformation *formats.JfrogResearchInformation) int {
+	if jfrogResearchInformation == nil {
+		return noJfrogResearchInformation
```

Comment:
can't we just return the defualt severity in case it is not existing? we want to also take care of the use case that a research severity is lower that normal severity of the other (without research info)